WordPress/openverse-frontend
# Add a fallback for broken provider image

## Fixes


Related to #914 by @obulat 

## Description


We are currently requesting provider media during the Playwright tests to render single image detail and audio detail pages. This creates flaky and slow tests. Until we create a solution to proxy those requests, we can block or mock the provider media requests entirely. 

Blocking the provider media requests would result in a broken image on the image detail page during Playwright tests. To prevent this, this PR adds the logic to replace the provider image with the thumbnail on load error. Since the thumbnails have the same dimensions as the provider image they allow us to test that the different dimensions are treated correctly). 

## Testing Instructions

Search for 'birds', and open the first image result. Block the requests to 'Flickr', and reload the image.
<img width="866" alt="Screen Shot 2022-06-08 at 11 17 21 AM" src="https://user-images.githubusercontent.com/15233243/172583355-2362936a-9e3a-476b-8e63-545a780e2e93.png">

 Visually, the page should be the same as before, but it should be using the 'thumbnail' from our API as the main image.
Also, try blocking the main thumbnail image. You should see the error image placeholder.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
